### DISCUSSION TITLE
Maximum current

### DISCUSSION THREAD

**Discussion by miccgn** [pixelit-project/PixelIt#99] [Ideas]
Hi,

have you ever considered adding a parameter for the maximum current that could be drawn by the matrix?

By default, the matrix library seems to limit the display to a value around 2 Ampere. At least, I wasn't able to drive the current higher than that - when I sent a fully white screen to the matrix, it became yellow. The more white pixels I removed, the brighter the remaining pixels became.

FastLED allows defining the maximum current drawn to be defined by
FastLED.setMaxPowerInMilliWatts(VALUE);

However, I have to admit that even with a value of 5V*6000mA=30000mW, I wasn't able to make the matrix draw more than 3A.

I could imagine a parameter might be handy, but, on the other hand, wrong settings might damage the power supply, or create a lot of heat, warming up the encasing.

**Comment by o0shojo0o** [comment]
Hi,
i have a 4 amp power supply on the matrix and I can go to 100% (255) brightness and with white pixels without a discoloration.  
But I must admit that I have not yet measured the real power consumption.

The FastLED.setMaxPowerInMilliWatts() is also used for WLED if I have seen it correctly. 
For my sake we can also offer this option, if it can help someone always gladly :)
